index.js

```diff
@@ -108,11 +109,17 @@ class Hyperdrive extends EventEmitter {
   _ready (cb) {
     const self = this
 
-    self.metadata.on('error', onerror)
-    self.metadata.on('append', update)
-    self.metadata.on('extension', extension)
-    self.metadata.on('peer-add', peeradd)
-    self.metadata.on('peer-remove', peerremove)
+    let listeners = {
+      'error': onerror,
+      'append': update,
+      'extension': extension,
+      'peer-add': peeradd,
+      'peer-remove': peerremove
+    }
```

Comment:
@andrewosh I'm wondering fs there any need in a checkout to actually listen for these events?

Reply:
well if they're not needed for a checkout, that's even better! no need for the weird `closeCheckout()` method. :D

Reply:
Yeah I'd agree that checkouts shouldn't be adding listeners for these events. We should prob do a check to see if the metadata matches what's passed in as an option (which is what a checkout does) before attaching listeners.